libgstd/libgstd_json.c

```diff
@@ -0,0 +1,204 @@
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "libgstd_json.h"
+
+#include <jansson.h>
+#include <stdlib.h>
+#include <string.h>
+
+static GstdStatus gstd_json_get_value (const char *json, const char *name,
+    json_t ** root, json_t ** out);
+
+static GstdStatus
+gstd_json_get_value (const char *json, const char *name, json_t ** root,
+    json_t ** out)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_error_t error;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (root != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  *root = json_loads (json, 0, &error);
+  if (!*root) {
+    ret = GSTD_LIB_OOM;
+    goto out;
+  }
+
+  *out = json_object_get (*root, name);
+  if (!*out) {
+    ret = GSTD_LIB_NOT_FOUND;
+    json_decref (*root);
+  }
+
+out:
+  return ret;
+}
+
+GstdStatus
+gstd_json_is_null (const char *json, const char *name, int *out)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_t *root;
+  json_t *data;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  ret = gstd_json_get_value (json, name, &root, &data);
+  if (GSTD_LIB_OK != ret) {
+    goto out;
+  }
+
+  *out = json_is_null (data);
+  json_decref (root);
+
+out:
+  return ret;
+}
+
+GstdStatus
+gstd_json_get_child_char_array (const char *json,
+    const char *array_name, const char *element_name, char **out[],
+    int *array_lenght)
+{
+  GstdStatus ret = GSTD_LIB_OK;
+  json_t *root;
+  json_t *array_data;
+  json_t *data, *name;
+  json_error_t error;
+  int i, j;
+
+  g_return_val_if_fail (json != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (array_name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (element_name != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (out != NULL, GSTD_LIB_NULL_ARGUMENT);
+  g_return_val_if_fail (array_lenght != NULL, GSTD_LIB_NULL_ARGUMENT);
+
+  root = json_loads (json, 0, &error);
+
+  array_data = json_object_get (root, array_name);
+
+  if (!json_is_array (array_data)) {
+    ret = GSTD_LIB_TYPE_ERROR;
+    goto unref;
+  }
+
+  *array_lenght = json_array_size (array_data);
+
+  /* Allocate enough memory for all names */
+  *out = malloc ((*array_lenght) * sizeof (char *));
+
+  for (i = 0; i < (*array_lenght); i++) {
+    const char *string;
+
+    data = json_array_get (array_data, i);
+    if (!json_is_object (data)) {
+      ret = GSTD_LIB_TYPE_ERROR;
+      goto clear_mem;
+    }
+
+    name = json_object_get (data, element_name);
+    if (!json_is_string (name)) {
+      ret = GSTD_LIB_TYPE_ERROR;
+      goto clear_mem;
+    }
+
+    string = json_string_value (name);
+
+    /**
+      * Jansson library frees memory after parent object is dereferenced,
```

Comment:
typo Json

Reply:
Jannson is a library which is used in this file

Reply:
Correct, my bad